cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -318,11 +318,15 @@ public OperationSnapshot createCluster(String name, String imageVersion, Map<Str
         .addServiceAccountScopes("https://www.googleapis.com/auth/cloud-platform")
         .setZoneUri(conf.getZone())
         .putAllMetadata(metadata);
+      String networkHostProjectId = Strings.isNullOrEmpty(conf.getNetworkHostProjectID()) ? projectId :
+        conf.getNetworkHostProjectID();
       // subnets are unique within a location, not within a network, which is why these configs are mutually exclusive.
       if (conf.getSubnet() != null) {
-        clusterConfig.setSubnetworkUri(conf.getSubnet());
+        clusterConfig.setSubnetworkUri(String.format("projects/%s/regions/%s/%s", networkHostProjectId,
+                                                     conf.getRegion(), conf.getSubnet()));
```

Comment:
does this have the same problem, where the dataproc region doesn't necessarily match the network's region?

Reply:
```
does this have the same problem, where the dataproc region doesn't necessarily match the network's region?
```

Not sure what problem this is. The setSubnetworkUri documentation says it can be a full path in the format above.

Reply:
conf.getRegion is the dataproc region. It's not necessarily the same as the network region right?

Reply:
Looking at this https://issues.cask.co/browse/CDAP-14376 it does seem like it is for dataproc. 

subnetwork are regional resource.  The provisioner currently takes subnetwork details without region. Is it correct? https://cloud.google.com/compute/docs/reference/rest/v1/subnetworks/get requires a region to be specified to fetch . subnetwork.

Reply:
Had some discussion on chat. The subnet region needs to be picked from zone. Will have to do the changes for that.